File: SportsStoreVb/SportsStoreVb.WebUI/Infrastructure/NinjectDependencyResolver.vb
Imports System.Web.Mvc
Imports Ninject
Imports SportsStoreVb.Domain
Imports SportsStoreVb.Domain.Concrete
Imports SportsStoreVb.Domain.SportsStoreVb.Domain.Concrete

Public Class NinjectDependencyResolver
    Implements IDependencyResolver
    ReadOnly kernel As IKernel

    Public Sub New(kernelParam As IKernel)
        kernel = kernelParam
        AddBindings()
    End Sub

    Public Function GetService(serviceType As Type) As Object Implements IDependencyResolver.GetService
        Return kernel.TryGet(serviceType)
    End Function

    Public Function GetServices(serviceType As Type) As IEnumerable(Of Object) Implements IDependencyResolver.GetServices
        Return kernel.GetAll(serviceType)
    End Function

    Private Sub AddBindings()
        ' Dim mock As New Mock(Of IProductRepository)
        ' mock.Setup(Function(m) m.Products).Returns(New List(Of Product) From {
        '                                               New Product() With{.Name = "Football", .Price = 25},
        '                                               New Product() With {.Name="Surf board", .Price = 179 },
        '                                               New Product() With{.Name = "Running shoes", .Price = 95 }
        '                                            })
        ' kernel.Bind(Of IProductRepository).ToConstant(mock.Object)
        kernel.Bind(Of IProductRepository).To(Of EFProductRepository)
    End Sub
End Class

File: SportsStoreCs/SportsStoreCs.WebUI/Infrastructure/NinjectDependencyResolver.cs
﻿namespace SportsStoreCs.WebUI.Infrastructure
{
    using System;
    using System.Collections.Generic;
    using System.Web.Mvc;
    using System.Web.WebSockets;
    using Domain.Concrete;
    using Domain.Entities;
    using Moq;
    using Ninject;
    using SportsStoreCs.Domain.Abstract;

    public class NinjectDependencyResolver : IDependencyResolver
    {
        private readonly IKernel kernel;

        public NinjectDependencyResolver(IKernel kernelParam)
        {
            kernel = kernelParam;
            AddBindings();
        }

        public object GetService(Type serviceType)
        {
            return kernel.TryGet(serviceType);
        }

        public IEnumerable<object> GetServices(Type serviceType)
        {
            return kernel.GetAll(serviceType);
        }

        public void AddBindings()
        {
            /*
            var mock = new Mock<IProductRepository>();
            mock.Setup(m => m.Products).Returns(new List<Product>
            {
                new Product {Name = "Football", Price = 25},
                new Product {Name = "Surf board", Price = 179},
                new Product {Name = "Running shoes", Price = 95}
            });
            kernel.Bind<IProductRepository>().ToConstant(mock.Object);
            */
            kernel.Bind<IProductRepository>().To<EFProductRepository>();
        }
    }
}